app/lib/link_checker/uri_checker/http_checker.rb

```diff
@@ -113,10 +113,12 @@ def call
         return add_problem(NoHost.new(from_redirect: from_redirect?))
       end
 
+      check_suspicious_domains
+      return report if report.problems.any?
```

Comment:
For some reason, `report.has_errors?` returns false here.

```
$ report
=> #<LinkChecker::UriChecker::Report:0x0000ffff91b30da8
 @problems=
  [#<LinkChecker::UriChecker::SuspiciousDomain:0x0000ffff947b3b28
    @from_redirect=false,
    @message="This link is hosted on a domain which is on our list of suspicious domains",
    @priority=18,
    @suggested_fix=nil,
    @summary="Suspicious Destination",
    @text_args={},
    @type=:danger>]>

[2] pry(#<LinkChecker::UriChecker::HttpChecker>)> report.has_errors?
=> false
```

Reply:
Ahhh, makes sense - thanks. I assumed `has_errors?` was a catch-all for everything.

Reply:
Yeah, it's a bit of a misleading name for a specific category of problems.  Like you, I'd initially have thought it would include any reported problems.